client/state/use-reducer.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { createElement, Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { addReducer, removeReducer } from 'state';
+
+/**
+ * Higher Order Component to add a reducer to the store's reducer list.
+ * TODO: Add this file to a place accessible by all extensions when it's ready.
+ *
+ * @param { string } reducerName The name for the reducer/key for redux state tree.
+ * @param { func } reducerFunc The function that takes (state, action) and returns state.
+ * @returns { func } A curried function that takes a component to wrap.
+ */
+export default function useReducer( reducerName, reducerFunc ) {
```

Comment:
I'd also like to note that if someone has a better idea as to where this HOC should live, I'm definitely open to that as well.